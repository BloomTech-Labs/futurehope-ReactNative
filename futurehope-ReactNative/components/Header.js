import React from 'react';
import { View, Text, StyleSheet, ShadowPropTypesIOS } from 'react-native';
import { NativeRouter, Route, Link, useHistory } from "react-router-native"
import Icon from 'react-native-vector-icons/FontAwesome'
import { widthPercentageToDP, heightPercentageToDP } from '../utils/PercenatageFix'

const Header = () => {
    const history = useHistory()
    function backHandler() {
        history.goBack()
    }

    return (
        <View style={styles.header}>
            <Icon style={styles.icon} name='arrow-left' onPress={backHandler} size={20} color='#ff9800' />
            <Text style={styles.text}>School in the Sky</Text>
            <Link style={styles.icon} to='/'><Icon name='home' color='#ff9800' size={20} /></Link>
        </View>
    )
}


const styles = StyleSheet.create({
    header: {
        flexDirection: 'row',
        height: heightPercentageToDP('9%'),
        paddingTop: heightPercentageToDP('4%'),
        paddingBottom: heightPercentageToDP('1%'),
        alignItems: 'center',
        backgroundColor: '#eee',
        justifyContent: 'space-between'
    },
    text: {

        fontSize: 25,
        textAlign: 'center'
    },
    icon: {
        marginLeft: widthPercentageToDP('4%'),
        marginRight: widthPercentageToDP('4%')
    }
});


export default Header;
