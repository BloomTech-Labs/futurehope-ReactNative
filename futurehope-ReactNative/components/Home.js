import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { NativeRouter, Route, Link } from "react-router-native"
import Icon from "react-native-vector-icons/FontAwesome";

const Home = ({ item, deleteItem }) => {

    return (
        <View style={styles.container}>
            <Link to='/notetaking'>
                <Text style={styles.text}><Icon name='file' size={40}/>  Notes</Text>
            </Link>
        </View>
    )
}

const styles = StyleSheet.create({
    text: {
        color: 'black',
        fontSize: 35,
        textAlign: 'center'
    },
    container: {
        height: 450,
        alignItems: 'center',
        justifyContent: 'center'
    }
});

export default Home;
