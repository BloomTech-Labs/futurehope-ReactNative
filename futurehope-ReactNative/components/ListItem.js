import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import {Link} from "react-router-native";
//import Icon from 'react-native-vector-icons/dist/FontAwesome';
import Icon from 'react-native-vector-icons/FontAwesome';

const ListItem = ({ item, deleteItem }) => {

    return (
        <TouchableOpacity style={styles.listItem}>
            <View style={styles.listItemView}>
               <Link to={`/noteedit/${item.id}`}><Text style={styles.listItemText}>{item.title}</Text></Link>
                <Icon onPress={() => deleteItem(item.id)} color='red' size={32} name='times-circle' />
            </View>
        </TouchableOpacity>
    )
}


const styles = StyleSheet.create({
    listItem: {
        padding: 15,
        backgroundColor: '#f8f8f8',
        borderBottomWidth: 1,
        borderColor: '#eee'
    },
    listItemView: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
    listItemText: {
        fontSize: 18
    }
});


export default ListItem;
