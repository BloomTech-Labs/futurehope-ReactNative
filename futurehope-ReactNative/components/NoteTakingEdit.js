import React, { useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableOpacity
} from "react-native";
import { useParams } from "react-router-native";
import {
  widthPercentageToDP,
  heightPercentageToDP
} from "../utils/PercenatageFix";

const EditItem = ({ items, editItem, edit, editChange }) => {
  const { id } = useParams();

  const [target, setTarget] = useState({});
  const [editNote, setEditNote] = useState(false);
  React.useEffect(() => {
    items.forEach(e => {
      if (e.id === id) {
        setTarget(e);
      }
    });
  }, [items]);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{target.title}</Text>
      {!target.content ? (
        <Text style={editNote ? styles.invis : styles.placeHolder}>
          Add details here...
        </Text>
      ) : (
        <Text style={editNote ? styles.invis : styles.contents}>
          {target.content}
        </Text>
      )}
      <TextInput
        placeholder={target.content || "Add details here..."}
        style={editNote ? styles.input : styles.invis}
        onChangeText={editChange}
        value={edit}
      />
      <TouchableOpacity
        style={editNote ? styles.button : styles.invis}
        onPress={() => {
          editItem(edit, target.id);
          setEditNote(false);
        }}
      >
        <Text style={styles.buttonText}>Set Details</Text>
      </TouchableOpacity>

      <TouchableOpacity
        style={editNote ? styles.invis : styles.editContentBtn}
        onPress={() => setEditNote(true)}
      >
        <Text style={styles.contentBtnText}>Edit Content</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: heightPercentageToDP("82%"),
    textAlign: "left",
    marginLeft: 10,
    marginRight: 10
  },
  invis: {
    display: "none"
  },
  top: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between"
  },
  title2: {
    fontSize: 30
  },
  title: {
    fontSize: 40,
    fontWeight: "bold",
    margin: 20
  },
  edit: {
    color: "#FFB23D",
    textAlign: "right",
    fontSize: 20,
    fontWeight: "bold"
  },
  placeHolder: {
    color: "grey"
  },
  input: {
    height: 60,
    padding: 8,
    fontSize: 16
  },
  button: {
    width: "100%",
    backgroundColor: "#FFB23D",
    position: "absolute",
    bottom: 10
  },
  buttonText: {
    color: "white",
    textAlign: "center",
    fontWeight: "bold",
    padding: 10,
    fontSize: 20
  },
  contents: {
    padding: 7,
    color: "grey",
    fontSize: 20
  },
  editContentBtn: {
    width: "100%",
    backgroundColor: "#FFB23D",
    position: "absolute",
    bottom: 10
  },
  contentBtnText: {
    color: "white",
    textAlign: "center",
    fontWeight: "bold",
    padding: 10,
    fontSize: 20
  }
});

export default EditItem;
