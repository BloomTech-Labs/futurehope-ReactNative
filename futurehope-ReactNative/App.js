import React, { useState, useEffect } from 'react';
import { View, Text, StyleSheet, FlatList, Alert, AsyncStorage } from 'react-native';
import uuid from 'react-native-uuid';
import Header from './components/Header';
import ListItem from './components/ListItem';
import AddItem from './components/AddItem';

const App = () => {

  const [items, setItems] = useState([
    { id: uuid.v4(), title: 'Note 1', content: '' },
    { id: uuid.v4(), title: 'Note 2', content: '' },
    { id: uuid.v4(), title: 'Note 3', content: '' },
    { id: uuid.v4(), title: 'Note 4', content: '' },
  ]);

  const deleteItem = (id) => {
    setItems(prevItems => {
      return prevItems.filter(item => item.id != id);
    });
  }

  const addItem = (title) => {
    setItems(prevItems => {
      return [{ id: uuid.v4(), title, content: '' }, ...prevItems];
    });
  };

  useEffect(() => {
    AsyncStorage.getItem("noteAppStorage").then(value => {
      setItems(JSON.parse(value))
    })
  }, [])

  useEffect(() => {
    AsyncStorage.setItem("noteAppStorage", JSON.stringify(items))
  }, [items])

  return (
    <View style={styles.container}>
      <Header />
      <AddItem addItem={addItem} />
      <FlatList
        data={items}
        renderItem={({ item }) => <ListItem item={item} deleteItem={deleteItem} />}
      />
    </View>
  )
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});


export default App;
