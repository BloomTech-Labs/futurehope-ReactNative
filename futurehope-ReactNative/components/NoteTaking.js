import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, FlatList, AsyncStorage } from 'react-native';
import ListItem from './ListItem';
import AddItem from './AddItem';
import uuid from 'react-native-uuid';
import { NativeRouter, Route, Link } from "react-router-native"
//import Icon from 'react-native-vector-icons/dist/FontAwesome';


const NoteTaking = () => {

    const [items, setItems] = React.useState([
        { id: uuid.v4(), title: 'Note 1', content: '' },
        { id: uuid.v4(), title: 'Note 2', content: '' },
        { id: uuid.v4(), title: 'Note 3', content: '' },
        { id: uuid.v4(), title: 'Note 4', content: '' },
    ]);

    const deleteItem = (id) => {
        setItems(prevItems => {
            return prevItems.filter(item => item.id != id);
        });
    }

    const addItem = (title) => {
        setItems(prevItems => {
            return [{ id: uuid.v4(), title, content: '' }, ...prevItems];
        });
    };

    React.useEffect(() => {
        AsyncStorage.getItem("noteAppStorage").then(value => {
            if (value) {
                setItems(JSON.parse(value))
            }
        })
    }, [])

    React.useEffect(() => {
        AsyncStorage.setItem("noteAppStorage", JSON.stringify(items))
    }, [items])

    return (
        <View>
            <AddItem addItem={addItem} />
            <FlatList
                data={items}
                renderItem={({ item }) => <ListItem item={item} deleteItem={deleteItem} />}
            />
        </View>
    )
}

export default NoteTaking