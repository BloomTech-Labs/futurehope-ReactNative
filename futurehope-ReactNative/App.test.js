import React from 'react';
import 'react-native';
import { render, fireEvent } from 'react-native-testing-library';

import App from './App';
import Header from './components/Header';
import Home from './components/Home';
import ListItem from './components/ListItem';
import NoteTaking from './components/NoteTaking';
import NoteTakingEdit from './components/NoteTakingEdit';


test('Header test against snapshot', () => {
    const tree = render(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
});


test('NoteTaking test against snapshot', () => {
    const tree = render(<NoteTaking />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('NoteTaking test against snapshot', () => {
    const tree = render(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Header name is correct', () => {
  const { getByText } = render(<Header />);
  const element = getByText('Future Hope');
});

// test('should invoke specified event', () => {
//    const onPressMock = jest.fn();
//    const { getByTestId } = render(<Home onPress={NoteTaking} />);
//
//    fireEvent(getByTestId('Link'), 'press');
//  });

test('debug.shallow', () => {
  expect(debug.shallow).toBe(debugShallow);
});
