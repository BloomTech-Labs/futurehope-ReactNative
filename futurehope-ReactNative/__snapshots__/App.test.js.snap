// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ff9800",
        "flexDirection": "row",
        "height": 75,
        "justifyContent": "space-between",
        "paddingLeft": 15,
        "paddingRight": 120,
        "paddingTop": 28,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text />
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "textAlign": "center",
        }
      }
    >
      Future Hope
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 450,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 35,
            "textAlign": "center",
          }
        }
      >
        <Text />
          Notes
      </Text>
    </View>
  </View>
</View>
`;

exports[`App test against snapshot 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#ff9800",
      "flexDirection": "row",
      "height": 75,
      "justifyContent": "space-between",
      "paddingLeft": 15,
      "paddingRight": 120,
      "paddingTop": 28,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "textAlign": "center",
      }
    }
  >
    Future Hope
  </Text>
</View>
`;

exports[`App test against snapshot 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 450,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35,
          "textAlign": "center",
        }
      }
    >
      <Text />
        Notes
    </Text>
  </View>
</View>
`;

exports[`App test against snapshot 4`] = `
<View>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Add item..."
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 16,
          "height": 60,
          "padding": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFB23D",
          "margin": 5,
          "opacity": 1,
          "padding": 9,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Add Note 
        <Text />
      </Text>
    </View>
  </View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Header test against snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ff9800",
      "flexDirection": "row",
      "height": 75,
      "justifyContent": "space-between",
      "paddingLeft": 15,
      "paddingRight": 120,
      "paddingTop": 28,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "textAlign": "center",
      }
    }
  >
    Future Hope
  </Text>
</View>
`;

exports[`Home test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 450,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35,
          "textAlign": "center",
        }
      }
    >
      <Text />
        Notes
    </Text>
  </View>
</View>
`;

exports[`NoteTaking test against snapshot 1`] = `
<View>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Add item..."
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 16,
          "height": 60,
          "padding": 8,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFB23D",
          "margin": 5,
          "opacity": 1,
          "padding": 9,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        Add Note 
        <Text />
      </Text>
    </View>
  </View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`NoteTaking test against snapshot 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 450,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID="Link"
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 35,
          "textAlign": "center",
        }
      }
    >
      <Text />
        Notes
    </Text>
  </View>
</View>
`;
